www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php

```diff
@@ -74,16 +74,31 @@
 
 $centreon = $oreon;
 
-isset($_GET["servicegroup"]) ? $id = htmlentities($_GET["servicegroup"], ENT_QUOTES, "UTF-8") : $id = "NULL";
-isset($_POST["servicegroup"]) ? $id = htmlentities($_POST["servicegroup"], ENT_QUOTES, "UTF-8") : $id = $id;
+$servicegroupId = filter_var(
+    $_GET['servicegroup'] ?? $_POST['servicegroup'] ?? null,
+    FILTER_VALIDATE_INT
+);
+
+// finding the user's allowed servicegroup
+$sgs = $centreon->user->access->getServiceGroupAclConf(null, 'broker');
+
+//checking if the user has ACL rights for this resource
+if (!$centreon->user->admin
+    && $servicegroupId !== null
+    && !array_key_exists($servicegroupId, $sgs)
+) {
+    echo '<div align="center" style="color:red">' .
+        '<b>You are not allowed to access this service group</b></div>';
+    exit();
+}
 
 /*
  * Getting time interval to report
  */
 $dates = getPeriodToReport();
 $start_date = htmlentities($_GET['start'], ENT_QUOTES, "UTF-8");
 $end_date = htmlentities($_GET['end'], ENT_QUOTES, "UTF-8");
-$servicegroup_name = getServiceGroupNameFromId($id);
+$servicegroup_name = getServiceGroupNameFromId($servicegroupId);
```

Comment:
CamelCase mandatory

Reply:
Refacto the whole code is out of my scope... I already modified the variable on which I worked on ;)